### PR TITLE
Tracker v2 proofs

### PR DESCRIPTION
**Pull request by gabriel**
@keybase/react-hackers

### CODE REVIEW

File: shared/common-adapters/text.desktop.js [hunk 1]
```diff
@@ -186,7 +186,7 @@ const lineCommon = {
   WebkitBoxOrient: 'vertical'
 }
 
-function lineClamp (lines: number) {
+export function lineClamp (lines: number):Object {
```
Comment (cjb) [inline shared/common-adapters/text.desktop.js]: should there be a space between : and Object?

File: shared/more/components-tracker.render.desktop.js [hunk 2]
```diff
@@ -3,20 +3,43 @@
...
+  {name: 'unreachable', type: 'rooter', id: 'unreachableId', meta: metaUnreachable, humanUrl: ''},
+  // TODO: Need to use state for checking; Refactor after nuking v1
+  {name: 'checking', type: 'rooter', id: 'checkingId', state: checking, humanUrl: ''},
+  {name: 'pending', type: 'rooter', id: 'pendingId', meta: metaPending, humanUrl: ''},
+  {name: 'upgraded', type: 'rooter', id: 'upgradedId', meta: metaUpgraded, humanUrl: ''}
 ]
 
 const propsDefault = {
-  closed: false,
+  closed: false, // TODO: Why?
```
Comment (cjb) [inline shared/more/components-tracker.render.desktop.js]: Not sure what this TODO means, maybe better to remove it or expand on it.

File: shared/tracker/bio.render.desktop.js [hunk 3]
```diff
@@ -226,10 +241,11 @@ const styles2 = {
     lineHeight: '21px',
     textAlign: 'center'
   },
-  followsYou: {
+  followLabel: {
     fontSize: 11,
     opacity: 0.6,
-    color: globalColors.grey1
+    color: globalColors.grey1,
```
Comment (cjb) [inline shared/tracker/bio.render.desktop.js]: I think we should use the new `globalColorsDZ2` for all colors in a `styles2`?

File: shared/tracker/proofs.render.desktop.js [hunk 4]
```diff
@@ -2,18 +2,179 @@
...
+    ...globalStyles.flexBoxColumn
+  },
+  row: {
+    ...globalStyles.flexBoxRow,
+    paddingTop: 12,
+    paddingLeft: 30,
+    paddingRight: 30,
+    alignItems: 'flex-start',
+    justifyContent: 'flex-start',
+    backgroundColor: globalColors.white,
```
Comment (cjb) [inline shared/tracker/proofs.render.desktop.js]: (another globalColorsDZ2)

File: shared/tracker/header.render.desktop.js [hunk 5]
```diff
@@ -36,7 +36,7 @@ export default class HeaderRender extends Component {
   }
 
   render2 (styles: Object) {
-    let headerStyle:Object = (this.props.currentlyFollowing ? styles.headerSuccess : styles.headerNormal)
+    let headerStyle:Object = ((this.props.currentlyFollowing && !this.props.changed) ? styles.headerSuccess : styles.headerNormal)
```
Comment (cjb) [inline shared/tracker/header.render.desktop.js]: (another space after colon)

**Comment by cjb** [comment]
:thumbsup: from me, anyone else in @keybase/react-hackers want to look?

File: shared/more/components-tracker.render.desktop.js (hunk 2 above)
Comment (MarcoPolo) [inline shared/more/components-tracker.render.desktop.js]: what's the question?

File: shared/tracker/proofs.render.desktop.js [hunk 6]
```diff
@@ -2,18 +2,179 @@
...
+  proofNameContainer: {
+    ...globalStyles.flexBoxRow,
+    flexWrap: 'wrap',
+    alignItems: 'flex-start'
+  },
+  proofName: {
+    // TODO: This doesn't ellipsis since the proof names are a single word,
+    // using word-wrap or text-wrap doesn't work either. We should try
+    // interspersing blank whitespace to wrap on
+    // ...lineClamp(1)
```
Comment (MarcoPolo) [inline shared/tracker/proofs.render.desktop.js]: What about [wbr](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/wbr) ?

File: shared/common-adapters/index.desktop.js [hunk 7]
```diff
@@ -8,3 +8,4 @@ export {default as Logo} from './logo'
 export {default as Text} from './text'
 export {default as Terminal} from './terminal'
 export {default as Icon} from './icon'
+export {lineClamp} from './text'
```
Comment (MarcoPolo) [inline shared/common-adapters/index.desktop.js]: I don't think this should be here, this isn't a component.

If you need line clamp you should import it from text directly

File: shared/common-adapters/text.desktop.js (hunk 1 above)
Comment (MarcoPolo) [inline shared/common-adapters/text.desktop.js]: Did you update the flow file?

File: shared/tracker/index.js [hunk 8]
```diff
@@ -58,18 +59,24 @@ class Tracker extends Component {
 
     const renderChangedTitle = this.props.trackerMessage
     const failedProofsNotFollowingText = `Some of ${this.props.username}'s proofs couldn't be verified. Track the working proofs?`
+    const currentlyFollowing = flags.tracker2 ? this.props.currentlyFollowing : !!this.props.lastTrack
+
+    const changed = !this.props.proofs.every(function (proof, index, ar) {
```
Comment (MarcoPolo) [inline shared/tracker/index.js]: This logic should be in the reducer, not the components

File: shared/tracker/index.js [hunk 9]
```diff
@@ -58,18 +59,24 @@ class Tracker extends Component {
...
-        currentlyFollowing: this.props.currentlyFollowing
+        currentlyFollowing: currentlyFollowing
       },
       headerProps: {
         reason: this.props.reason,
         onClose: () => this.props.onClose(this.props.username),
         trackerState: this.props.trackerState,
-        currentlyFollowing: this.props.currentlyFollowing
+        currentlyFollowing: currentlyFollowing,
+        changed: changed
```
Comment (MarcoPolo) [inline shared/tracker/index.js]: if the key and value are the same you can just use the var name. so like `{currentlyFollowing, changed}`

File: shared/common-adapters/text.desktop.js (hunk 1 above)
Comment (gabriel) [inline shared/common-adapters/text.desktop.js]: Fixed

File: shared/more/components-tracker.render.desktop.js [hunk 10]
```diff
@@ -3,16 +3,39 @@
...
-import {normal, revoked} from '../constants/tracker'
 import flags from '../util/feature-flags'
+import {normal, checking, error} from '../constants/tracker'
+import {metaNew, metaUpgraded, metaUnreachable, metaPending, metaDeleted} from '../constants/tracker'
+
+const proofGithub = {name: 'githubuser', type: 'github', id: 'githubId', humanUrl: 'github.com', profileUrl: 'http://github.com'}
+const proofTwitter = {name: 'twitteruser', type: 'twitter', id: 'twitterId', humanUrl: 'twitter.com', profileUrl: 'http://twitter.com'}
+const proofWeb = {name: 'thelongestdomainnameintheworldandthensomeandthensomemoreandmore.com', type: 'web', id: 'webId', humanUrl: 'thelongestdomainnameintheworldandthensomeandthensomemoreandmore.com'}
+const proofHN = {name: 'pg', type: 'hackernews', id: 'hnId', humanUrl: 'news.ycombinator.com', profileUrl: 'http://news.ycombinator.com'}
+const proofRooter = {name: 'roooooooter', type: 'rooter', id: 'rooterId', humanUrl: ''}
```
Comment (MarcoPolo) [inline shared/more/components-tracker.render.desktop.js]: lol

**Comment by gabriel** [comment]
ok I think I fixed everything...

currentlyFollowing and changed are temporary... will change to use tracker states once @cjb's stuff goes in

File: shared/tracker/bio.render.desktop.js [hunk 11]
```diff
@@ -269,21 +268,21 @@ const followBadgeStyles = {
...
     top: 52,
     left: 55,
-    background: globalColors.green
+    background: globalColorsDZ2.green2
   },
   notFollowsYou: {
     top: 52,
     left: 55,
-    background: globalColors.grey3
+    background: '#ccc'
```
Comment (MarcoPolo) [inline shared/tracker/bio.render.desktop.js]: Is the color you want not in globalColorsDZ2?

File: shared/tracker/header.render.desktop.js [hunk 12]
```diff
@@ -36,8 +36,8 @@ export default class HeaderRender extends Component {
   }
 
   render2 (styles: Object) {
-    let headerStyle:Object = (this.props.currentlyFollowing ? styles.headerSuccess : styles.headerNormal)
-    let headerTextStyle:Object = styles.headerTextNormal
+    let headerStyle: Object = ((this.props.currentlyFollowing && !this.props.changed) ? styles.headerSuccess : styles.headerNormal)
+    let headerTextStyle: Object = styles.headerTextNormal
```
Comment (MarcoPolo) [inline shared/tracker/header.render.desktop.js]: fwiw, I don't think you have to annotate these, it infers the type object.

File: shared/tracker/bio.render.desktop.js (hunk 11 above)
Comment (gabriel) [inline shared/tracker/bio.render.desktop.js]: not in style guide

File: shared/tracker/header.render.desktop.js (hunk 12 above)
Comment (gabriel) [inline shared/tracker/header.render.desktop.js]: fixed

**Comment by MarcoPolo** [comment]
:+1:

**Pull request merged**